### PR TITLE
don't un-json `extra` if it's the empty string

### PR DESCRIPTION
**Pull request by simbabque**
The default value for the extra field is the empty string, because it's an empty text box. But it's supposed to be JSON and it was always decoded from JSON, which produced an exception.

### CODE REVIEW

**Comment by mickeyn** [comment]
tidy test fails over some weird edge-case we struggled with before.
please revert the tidy changes and leave only yours, otherwise it will keep failing on travis.

File: lib/MetaCPAN/Web/Controller/Account.pm [hunk 1]
```diff
@@ -52,21 +52,24 @@ sub profile : Local : Args(0) {
     return unless ( $req->method eq 'POST' );
 
     my $data = $author;
-    $data->{blog} = $req->param('blog.url')
+    $data->{blog}
```
Comment (mickeyn) [inline lib/MetaCPAN/Web/Controller/Account.pm]: this is a weird tidyall issue (probably version mismatch), this change fails on Travis.

File: lib/MetaCPAN/Web/Controller/Account.pm [hunk 2]
```diff
@@ -52,21 +52,24 @@ sub profile : Local : Args(0) {
...
+        = $req->param('blog.url')
         ? [
         pairwise { { url => $a, feed => $b } }
         @{ [ $req->param('blog.url') ] },
         @{ [ $req->param('blog.feed') ] }
         ]
         : undef;
-    $data->{donation} = $req->param('donation.name')
+    $data->{donation}
```
Comment (mickeyn) [inline lib/MetaCPAN/Web/Controller/Account.pm]: same

File: lib/MetaCPAN/Web/Controller/Account.pm [hunk 3]
```diff
@@ -52,21 +52,24 @@ sub profile : Local : Args(0) {
...
+        = $req->param('donation.name')
         ? [
         pairwise { { name => $a, id => $b } }
         @{ [ $req->param('donation.name') ] },
         @{ [ $req->param('donation.id') ] }
         ]
         : undef;
-    $data->{profile} = $req->param('profile.name')
+    $data->{profile}
```
Comment (mickeyn) [inline lib/MetaCPAN/Web/Controller/Account.pm]: same

**Comment by mickeyn** [comment]
thanks
